hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/HoodieSparkCompactor.java

```diff
@@ -43,19 +48,24 @@ public HoodieSparkCompactor(BaseHoodieWriteClient<T, JavaRDD<HoodieRecord<T>>, J
   }
 
   @Override
-  public void compact(HoodieInstant instant) throws IOException {
+  public void compact(HoodieInstant instant) {
     LOG.info("Compactor executing compaction " + instant);
     SparkRDDWriteClient<T> writeClient = (SparkRDDWriteClient<T>) compactionClient;
-    JavaRDD<WriteStatus> res = writeClient.compact(instant.getTimestamp());
-    this.context.setJobStatus(this.getClass().getSimpleName(), "Collect compaction write status");
-    long numWriteErrors = res.collect().stream().filter(WriteStatus::hasErrors).count();
+    HoodieWriteMetadata<JavaRDD<WriteStatus>> compactionMetadata = writeClient.compact(instant.getTimestamp());
+    List<HoodieWriteStat> writeStats = compactionMetadata.getCommitMetadata().get()
+        .getPartitionToWriteStats()
+        .values()
```

Comment:
Is there any possibility we add a util method named `HoodieCommitMetadata#getWriteStats` to wrap these stream lines ? I saw the similar code invoked in several places.

Reply:
Sure, changed.